Version 1.1.0: ".zip" file extension support + 4 bug fixes

MichaelCurrie opened the pull request:
Features:

- #82 - "zip" file extension support

Bug fixes:

- #81 - setup instructions broken for Ubuntu 
- #86 - pip package missing schema file 
- #87 - bug with `__add__` operation and multiple worms
- #92 - `save_to_file` doesn't order dictionaries nested from lists

MichaelCurrie commented:
@Ichoran please review and merge if you think it looks good.

Ichoran commented:
@MichaelCurrie - I think this looks good, but could you fix the Travis tests?  It doesn't like the format of the `pep` command, apparently, even though it's happy with running the tests themselves.

Ichoran commented on WCON_format.md:
```diff
@@ -397,3 +397,13 @@ In this example, we imagine a directory containing files labelled `filename_0.wc
+
+- If an archive is loaded containing zero files, an error is raised.
+- If an archive contains exactly one file, this file is loaded.
+- If an archive contains more than one file, one file (not necessarily the first) is selected for loading.  If this file contains links via the "files" object in the specification (see above), then the other files in the archive may also end up loaded as well.  For this reason chunks may be archived together into one `.wcon.zip` file for convenience.
```
Can we make this less conditional?  Either the files _will_ get loaded (when linked), or the interface must let you select which file to load.  (Or both.)  A "maybe" isn't enough to allow chunks to be archived together.

MichaelCurrie replied:
Good point.  

> then the other files in the archive may also end up loaded as well.

should be changed to 

> then all linked files in the archive will be loaded.

Fixed with 9c3ba26

Ichoran commented on src/Python/wcon/wcon_schema.json:
```diff
@@ -1,164 +0,0 @@
-{
-    "$schema": "http://json-schema.org/draft-04/schema#",
-    "title": "Worm tracker Commons Object Notation (WCON)",
```
Why is the schema deleted?

MichaelCurrie replied:
See #86: basically `wcon_schema.json` was having to be stored in two places in the repo, in the root folder and also at `src/Python/wcon/`, but I discovered that the packager setup file `setup.py` lets you specify files outside the package folder to be packaged alongside during the packaging process.  So we no longer need to have two different identical files `wcon_schema.json`.  It's still stored in the root folder.

MichaelCurrie commented:
Since I've addressed your issues I'll merge this so I can fix @cheelee 's PR issue while I have the time right now :)  Thanks for the feedback @Ichoran !
